ui/pages/confirmations/components/confirm/info/approve/hooks/use-approve-token-simulation.ts

```diff
@@ -15,6 +15,15 @@ function isSpendingCapUnlimited(decodedSpendingCap: number) {
   return decodedSpendingCap >= UNLIMITED_THRESHOLD;
 }
 
+function toNonScientificString(num: number): string {
```

Comment:
Note for other reviewers, this function was ported here from this other recent PR: https://github.com/MetaMask/metamask-extension/pull/27992

This will create a conflict when we merge this back into `develop`; we'll need to update this file to import this function instead if using it inlined here (i.e. we'll need to accept the `develop` version when it conflicts, as we typically would).